updater/lib/dependabot/file_fetcher_job.rb

```diff
@@ -13,7 +13,9 @@ def perform_job
       begin
         connectivity_check if ENV["ENABLE_CONNECTIVITY_CHECK"] == "1"
         clone_repo_contents
-        @base_commit_sha = file_fetcher.commit || "unknown"
+        @base_commit_sha = file_fetcher.commit
+        raise "base commit SHA not found" unless @base_commit_sha
```

Comment:
Do we need to handle this new error anywhere?

Reply:
No, I want it to fall through and get logged as an [unknown_error](https://github.com/dependabot/dependabot-core/blob/f7d8921bc2df495a58d179b09fd3892123566e48/updater/lib/dependabot/file_fetcher_job.rb#L156-L160) so I can continue to chase it down more.